File: JavaScript/8js 1/arr.js
/*
|--------------------------------------------------------------------------
| Filling an array (part 1)
|--------------------------------------------------------------------------
|
*/

function arr(N) {
   let res = [];
   for (let i = 0; i < N; i++) {
      res.push(i);
   }
   return res;
}


const arr = n => n ? [...Array(n).keys()] : [];


function arr(n) {
   return Array.from({
      length: n
   }, (_, i) => i);
}


function arr(n) {
   return [...Array(n)].map((_, i) => i);
}


console.log(arr(5));

File: Python/8py 1/arr.py
"""
|--------------------------------------------------------------------------
| Filling an array (part 1)
|--------------------------------------------------------------------------
|
"""

def arr(n=0): 
   res = []
   for i in range(n):
      res.append(i)
   return res


def arr(n=0): 
   return list(range(n))


def arr(n=0): 
   return [i for i in range(n)]


def arr(n=0): 
   return [*range(n)]


print(arr(4), [0,1,2,3])
print(arr(0), [])
print(arr(), [])